src/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -337,109 +332,183 @@ contract BLSRegistryCoordinatorWithIndices is EIP712, Initializable, IBLSRegistr
+         * specified in `operatorKickParams`
+         */
+
+        uint256 kickIndex = 0;
+        for (uint256 i = 0; i < quorumNumbers.length; i++) {
+            uint8 quorumNumber = uint8(quorumNumbers[i]);
+            
+            OperatorSetParam memory operatorSetParams = _quorumParams[quorumNumber];
+            bool exceedsOperatorCapacity = numOperatorsPerQuorum[i] > operatorSetParams.maxOperatorCount;
+
+            /**
+             * If we exceed the max operator count for this quorum, we need to replace an existing operator
+             * with a new one.
+             */
+            if (exceedsOperatorCapacity) {
+                require(performChurn, "BLSRegistryCoordinatorWithIndices._registerOperator: churn required for overfilled quorum");
+                
+                // Validate that `operatorToRegister` can replace the operator specified in the kick params
+                _validateChurn({
+                    quorumNumber: quorumNumber,
+                    totalQuorumStake: totalStakes[i],
+                    newOperator: operatorToRegister,
+                    newOperatorStake: registeredStakes[i],
+                    kickParams: operatorKickParams[i],
+                    setParams: operatorSetParams
+                });
+
+                _deregisterOperator(operatorKickParams[i].operator, quorumNumbers[i:i+1]);
+                kickIndex++;
+            }
         }
+    }
 
-        emit OperatorSocketUpdate(operatorId, socket);
-        return (numOperatorsPerQuorum, operatorId);
+    function _validateChurn(
+        uint8 quorumNumber, 
+        uint96 totalQuorumStake,
+        address newOperator, 
+        uint96 newOperatorStake,
+        OperatorKickParam memory kickParams, 
+        OperatorSetParam memory setParams
+    ) internal view {
+        address operatorToKick = kickParams.operator;
+        bytes32 idToKick = _operatorInfo[operatorToKick].operatorId;
+        require(_operatorInfo[operatorToKick].status == OperatorStatus.REGISTERED, "BLSRegistryCoordinatorWithIndices._validateChurn: cannot remove unregistered operator");
+        require(newOperator != operatorToKick, "BLSRegistryCoordinatorWithIndices._validateChurn: cannot churn self");
+        require(kickParams.quorumNumber == quorumNumber, "BLSRegistryCoordinatorWithIndices._validateChurn: quorumNumber not the same as signed");
+
+        // Get the target operator's stake and check that it is below the kick thresholds
+        uint96 operatorToKickStake = stakeRegistry.getCurrentOperatorStakeForQuorum(idToKick, quorumNumber);
+        require(
+            newOperatorStake > _individualKickThreshold(operatorToKickStake, setParams),
+            "BLSRegistryCoordinatorWithIndices._validateChurn: incoming operator has insufficient stake for churn"
+        );
+        require(
+            operatorToKickStake < _totalKickThreshold(totalQuorumStake, setParams),
```

Comment:
extremely minor nit, but should these maybe be non-strict inequalities?

Reply:
Question for @gpsanant 

I feel like maybe the top one makes sense as a non-strict inequality, but the bottom doesnt?

Reply:
lets keep it strict for now, that's what's in the offchain code